src/apic-extension/azext_apic_extension/tests/latest/test_integration_commands.py

```diff
@@ -0,0 +1,70 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import jmespath
+import collections
+from azure.cli.testsdk import ScenarioTest, ResourceGroupPreparer
+from azure.cli.testsdk.checkers import JMESPathCheck
+from azure.cli.testsdk.exceptions import JMESPathCheckAssertionError
+from .utils import ApicServicePreparer, ApimServicePreparer
+from .constants import TEST_REGION, USERASSIGNED_IDENTITY
+
+# override the JMESPathCheck class to support checking multiple possible values as a list
+class JMESPathCheckAny(JMESPathCheck):
+    def __init__(self, query, expected_results, case_sensitive=True):
+        super().__init__(query, expected_results, case_sensitive)
+        if not isinstance(expected_results, list):
+            raise ValueError("expected_results should be a list of possible values")
+
+    def __call__(self, execution_result):
+        json_value = execution_result.get_output_in_json()
+        actual_value = jmespath.search(self._query, json_value, jmespath.Options(collections.OrderedDict))
+        if self._case_sensitive:
+            if actual_value not in [result for result in self._expected_result]:
+                raise JMESPathCheckAssertionError(self._query, self._expected_result, actual_value, execution_result.output)
+        else:
+            if actual_value.lower() not in [result.lower() for result in self._expected_result]:
+                raise JMESPathCheckAssertionError(self._query, self._expected_result, actual_value, execution_result.output)
+
+class IntegrationCommandTests(ScenarioTest):
+    # override the check method to support checking multiple possible values
+    def check(self, query, expected_results, case_sensitive=True):
+        query = self._apply_kwargs(query)
+        expected_results = self._apply_kwargs(expected_results)
+        
+        if isinstance(expected_results, list):
+            return JMESPathCheckAny(query, expected_results, case_sensitive)
+        else:
+            return JMESPathCheck(query, expected_results, case_sensitive)
+
+    # TODO: change the location to TEST_REGION when the APIC resource provider is available in all regions
+    @ResourceGroupPreparer(name_prefix="clirg", location="centraluseuap", random_name_length=32)
+    @ApicServicePreparer()
+    @ApimServicePreparer()
+    def test_integration_create_apim(self):
+        if self.is_live:
```

Comment:
does this mean this test case will not be executed in playback mode?

Reply:
after some tries, I found there will be conflict that in playback mode, the sub is mocked as 00000-...-000, but the msi-id should be a real one.